File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        //We can flip at most k nums 
        int i=0;
        int j=0;
        int ans=0;
        int flip_count=0;
        int n = nums.size();
        while(j<n) {
            if (nums[j]==0)  flip_count+=1;
            while(flip_count>k) {
                if(nums[i]==0) flip_count-=1;
                i++;
            }

            if(flip_count<=k) ans = max(ans,j-i+1);
            j++;
        }
        return ans;
        
    }
};

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l,flipCount = 0,0
        answer = 0
        for r,num in enumerate(nums):
            if num==0:
                flipCount+=1
            while flipCount>k:
                if nums[l]==0:
                    flipCount-=1
                l+=1
            answer = max(answer,r-l+1)
        return answer

        